src/optimizer/rule/RemoveUselessProjectRule.cpp

```diff
@@ -0,0 +1,79 @@
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "optimizer/rule/RemoveUselessProjectRule.h"
+
+#include "optimizer/OptContext.h"
+#include "optimizer/OptGroup.h"
+#include "planner/PlanNode.h"
+#include "planner/Query.h"
+
+using nebula::graph::PlanNode;
+using nebula::graph::QueryContext;
+
+namespace nebula {
+namespace opt {
+
+std::unique_ptr<OptRule> RemoveUselessProjectRule::kInstance =
+    std::unique_ptr<RemoveUselessProjectRule>(new RemoveUselessProjectRule());
+
+RemoveUselessProjectRule::RemoveUselessProjectRule() {
+    RuleSet::QueryRules().addRule(this);
+}
+
+const Pattern &RemoveUselessProjectRule::pattern() const {
+    static Pattern pattern = Pattern::create(
+        graph::PlanNode::Kind::kProject, {Pattern::create(graph::PlanNode::Kind::kAggregate)});
+    return pattern;
+}
+
+StatusOr<OptRule::TransformResult> RemoveUselessProjectRule::transform(
+    OptContext *octx,
+    const MatchedResult &matched) const {
+    auto qctx = octx->qctx();
+    auto* projGroupNode = matched.node;
+    DCHECK_EQ(projGroupNode->node()->kind(), PlanNode::Kind::kProject);
+    auto* oldProjNode = static_cast<const graph::Project*>(projGroupNode->node());
+    std::vector<YieldColumn*> cols = oldProjNode->columns()->columns();
+    auto allVarPropExpr = true;
+    for (auto* col : cols) {
+        if (col->expr()->kind() != Expression::Kind::kVarProperty) {
+            allVarPropExpr = false;
+            break;
+        }
+    }
+    if (!allVarPropExpr) {
+        return TransformResult::noTransform();
+    }
```

Comment:
Could move this logics to `match` interface.

Reply:
ACK.